src/main/java/org/broadinstitute/dsm/db/KitType.java

```diff
@@ -88,4 +95,43 @@ public KitType(int kitId, String name, String displayName, boolean manualSentTra
         logger.info("Found " + kitTypes.size() + " kitTypes ");
         return kitTypes;
     }
+
+    public static List<String> getUploadReasons(@NonNull String realm){
+        SimpleResult results = inTransaction((conn) -> {
+            List<String> uploadReasons = new ArrayList<>();
+            SimpleResult dbVals = new SimpleResult();
+            String query = SQL_SELECT_UPLOAD_REASONS;
+
+            try (PreparedStatement stmt = conn.prepareStatement(query)) {
+                stmt.setString(1, realm);
+                try (ResultSet rs = stmt.executeQuery()) {
+                    while (rs.next()) {
+
+                        String reasons = rs.getString(DBConstants.UPLOAD_REASONS);
+                        if (StringUtils.isBlank(reasons)) {
+                            dbVals.resultValue = null;
+                            return dbVals;
+                        }
+                        uploadReasons = Arrays.asList(new Gson().fromJson(reasons, String[].class));
```

Comment:
Storing the reasons as an inline bit of json will make edits more challenging down the road.  How about making a separate join table that has one row per reason?

Reply:
that is just for the kit_request_setting for study staff at the upload to select. the actual reason for a kit request will be in the kit_request table. so I say it is fine for the configuration of a study